go.mod

```diff
@@ -1,21 +1,22 @@
 module github.com/kubeflow/pipelines
 
-go 1.22.0
+go 1.23.1
```

Comment:
just a heads up, we'll need to update to go 1.23 in konflux within the rhds repository as well, the latest crypto libraries coming in as part of this update has a hard  dependency on 1.23,
example: https://github.com/red-hat-data-services/data-science-pipelines/blob/main/backend/Dockerfile